openedx/core/djangoapps/content/course_structures/migrations/0001_initial.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+import datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
+
+    def forwards(self, orm):
+        # Adding model 'CourseStructure'
+        db.create_table('course_structures_coursestructure', (
+            ('id', self.gf('django.db.models.fields.AutoField')(primary_key=True)),
+            ('created', self.gf('model_utils.fields.AutoCreatedField')(default=datetime.datetime.now)),
+            ('modified', self.gf('model_utils.fields.AutoLastModifiedField')(default=datetime.datetime.now)),
+            ('course_id', self.gf('xmodule_django.models.CourseKeyField')(max_length=255, db_index=True)),
+            ('version', self.gf('django.db.models.fields.CharField')(max_length=255, null=True, blank=True)),
+            ('structure_json', self.gf('django.db.models.fields.TextField')()),
+        ))
+        db.send_create_signal('course_structures', ['CourseStructure'])
+
+        # Adding unique constraint on 'CourseStructure', fields ['course_id', 'version']
+        db.create_unique('course_structures_coursestructure', ['course_id', 'version'])
```

Comment:
If we're storing many versions for a course, we'll want an index to find the most recently published version. That might mean a new field entirely (e.g. `published`) that's sent as a param to the celery task. Then you could do a `db.create_index()` on `course_id` and `-published. It's slightly different from `modified` since there might be a gap between when publishes happen and when the worker sees the messages.